add a lot of upgrade contents (#542)
wireapp/wire-server-deploy: offline/upgrading.md

Obtain the latest airgrap artifact for wire-server-deploy. Please contact us to get it for now. We are
working on publishing a list of airgap artifacts.

## Clean up enough disk space to operate:

### AdminHost

> prune old containers that are generated during our 'd' invocations:

Reviewer: ```suggestion
Prune old containers that are generated during our 'd' invocations:
```

> prune old containers that are generated during our 'd' invocations:

Reviewer: ```suggestion
prune old containers that are generated during our `d` invocations:
```

```

> df

Reviewer: ```suggestion
df -h
```

`-h` is for *human readable*.

sudo docker container prune
```

### Kubernetes hosts:

> Remove wire-server images from two releases ago, or from the current release that we know are unused. for instance, 

Reviewer: ```suggestion
Remove wire-server images from two releases ago, or from the current release that we know are unused. For instance, 
```

```
sudo docker image ls

> sudo docker image ls | grep -E "^quay.io/wire/" | grep "2.106.0" | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"

Reviewer: ```suggestion
VERSION="2.106.0"
sudo docker image ls | grep -E "^quay.io/wire/" | grep $VERSION | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"
```

Give a name to the magic number. Not all customers might know our release naming schema and immediately recognize it.

```

> if you are running a DMZ deployment, prune the old wire-server images and their dependencies on the SFT kubernetes hosts...

Reviewer: ```suggestion
If you are running a DMZ deployment, prune the old wire-server images and their dependencies on the SFT kubernetes hosts...
```

```
sudo docker image ls | grep -E "^quay.io/wire/(team-settings|account|webapp|namshi-smtp)" | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"
sudo docker image ls | grep -E "^bitnami/redis" | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"
sudo docker image ls | grep -E "^airdock/fake-sqs" | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"
sudo docker image ls | grep -E "^localstack/localstack" | sed "s/.*[ ]*\([0-9a-f]\{12\}\).*/sudo docker image rm \1/"
sudo docker image rm 
```

> and remove SFT images from the Wire kubernetes.

Reviewer: ```suggestion
and remove SFT images from the Wire Kubernetes.
```

## Comparing the inventory

> copy `ansible/inventory/offline/99-static` to `ansible/inventory/offline/hosts.ini`.

Reviewer: ```suggestion
Copy `ansible/inventory/offline/99-static` to `ansible/inventory/offline/hosts.ini`.
```

Compare the inventory from your old install to the inventory of your new install.

> Here you will describe the topology of your offline deploy.  There's instructions in the comments on how to set

Reviewer: ```suggestion
Here you will describe the topology of your offline deploy. There are instructions in the comments on how to set
```

```
They should all report ready.

If you are worried about disk space, removing unneeded images could be performed here.

## Upgrading wire-server using helm

### Upgrading non-wire components:

Copy your external service definition values into place.
```

> ```

Reviewer: Looks like some content is missing here... :thinking:

First, upgrade the external service definitions, as those rarely change.
```
d helm upgrade cassandra-external ./charts/cassandra-external/ --values ./values/cassandra-external/values.yaml
d helm upgrade elasticsearch-external ./charts/elasticsearch-external/ --values ./values/elasticsearch-external/values.yaml
d helm upgrade minio-external ./charts/minio-external/ --values ./values/minio-external/values.yaml
```

Next, upgrade the internal non-wire services.
```
d helm upgrade fake-aws ./charts/fake-aws/ --values ./values/fake-aws/prod-values.example.yaml
d helm upgrade databases-ephemeral ./charts/databases-ephemeral/ --values ./values/databases-ephemeral/values.yaml
d helm upgrade reaper ./charts/reaper/
```

> Finally, upgrade demo-smtp. note that you may have to look for a values.yaml, instead of a prod-values.example.yaml file, in the case that demo-smtp was configured. if it's not present, use prod-values.example.yaml.

Reviewer: ```suggestion
Finally, upgrade demo-smtp. Note that you may have to look for a `values.yaml`, instead of a `prod-values.example.yaml` file, in the case that demo-smtp was configured. If it's not present, use `prod-values.example.yaml`.
```

```
d helm upgrade demo-smtp ./charts/demo-smtp/ --values ./values/demo-smtp/values.yaml
```

### Upgrading the NginX Ingress
```
d helm upgrade nginx-ingress-controller ./charts/nginx-ingress-controller/
d helm upgrade nginx-ingress-services ./charts/nginx-ingress-services/ --values ./values/nginx-ingress-services/values.yaml  --values ./values/nginx-ingress-services/secrets.yaml
```

### Upgrading Wire itsself

> inspect your values and secrets files with diff comparing them to the new defaults.

Reviewer: ```suggestion
Inspect your `values.yaml` and `secrets.yaml` files with diff comparing them to the new defaults.
```

kubenode4 node_labels="wire.com/role=sftd" node_annotations="{'wire.com/external-ip': 'XXXX'}"
```

> If the node does not know its own public IP (e.g. becuase it's behind NAT) then you should also set

Reviewer: ```suggestion
If the node does not know its own public IP (e.g. because it's behind NAT) then you should also set
```

kubenode3 node_labels="{'wire.com/role': 'sftd'}" node_annotations="{'wire.com/external-ip': 'XXXX'}"
```

> you may also want to look at the output of 'd kubectl get node' for each node, and to see if the node_sabel and the annotations are in order.

Reviewer: ```suggestion
You may also want to look at the output of `d kubectl get node` for each node, and to see if the `node_sabel` attribute and the annotations are in order.
```
